### ISSUE TITLE
Automatically set Alluxio hostname for EMR applications

### ISSUE THREAD

**Issue by apc999** [Alluxio/alluxio#9476]
**Is your feature request related to a problem? Please describe.**

Automatically set property `alluxio.master.hostname` based on EMR master address (which will be different across different EMR runs)  for EMR apps like Spark, rather than asking users to input the master addr manually in Hivemetastore or Glue. This is because every time when launching a new EMR cluster, this address is different so they have to check to use Presto or Spark on EMR with Alluxio.

**Describe the solution you'd like**

Modify the bootstrap and set the `alluxio.master.hostname` property in corresponding `core-site.xml` file or other configuration files for EMR apps.

**Describe alternatives you've considered**

**Urgency**
Low

**Additional context**

**Comment by apc999** [comment]
I tested with the latest EMR script and it works as what we proposed in this ticket. 
However, this is because the current EMR bootstrap script is installing Alluxio into `/opt/alluxio` and by default, an alluxio java client will search `alluxio-site.properties` inside `/opt/alluxio/conf`. So it works.

However, if someone is modifying the installation dir to a different location like `/opt/alluxio/v2.0`, it will break. We need to ensure one of the following
(1) the installation path is `/opt/alluxio`
(2) make sure we add `alluxio.master.hostname` to every EMR apps as a Java Opt
(3) respect some env var in Java client to get the setting.
